core/events/block_events.js

```diff
@@ -247,12 +247,16 @@ Blockly.Events.Create = function(opt_block) {
     this.recordUndo = false;
   }
 
-  if (opt_block.workspace.rendered) {
-    this.xml = Blockly.Xml.blockToDomWithXY(opt_block);
-  } else {
-    this.xml = Blockly.Xml.blockToDom(opt_block);
-  }
+  this.xml = Blockly.Xml.blockToDomWithXY(opt_block);
   this.ids = Blockly.Events.getDescendantIds(opt_block);
+
+  /**
+   * JSON representation of the block that was just created.
+   * @type {!Blockly.serialization.blocks.State}
+   * @private
+   */
+  this.blockJson_ = /** @type {!Blockly.serialization.blocks.State} */
+      (Blockly.serialization.blocks.save(opt_block, {addCoordinates: true}));
```

Comment:
As discussed offline, I think I would prefer if this was public because it makes it more consistent with the delete event and with the xml system.